### DISCUSSION TITLE
I'm using Dagre Js for autolayout and the new nodes that I'm adding are not aligned straight along the y-axis.

### DISCUSSION THREAD

**Discussion by naveenpaul** [xyflow/xyflow#4250] [Layouting]
### Describe the Bug

I'm using Dagre Js for autolayout and the new nodes that I'm adding are not aligned straight along the y-axis. 
How do I reposition these nodes properly?

### Steps to Reproduce the Bug or Issue

Attaching screenshot

### Expected behavior

Attaching screenshots

### Screenshots or Videos

Current Behavior: 

<img width="590" alt="Screenshot 2022-09-01 at 9 51 15 AM" src="https://user-images.githubusercontent.com/1685291/187831655-8bf406aa-cbfd-42b5-a2b3-3c848f6b6fec.png">

Expected:
![image](https://user-images.githubusercontent.com/1685291/187831718-f848cdd9-4b89-474c-9e4d-17a1c7e418b5.png)

### Platform

- OS: macOS, 
- Browser: Chrome

**Comment by chrtze** [comment]
Maybe the problem here is that the last node has a different size than the others? You can try passing the actual width and height of the nodes to dagre:

``` 
nodes.forEach((node) => {
    dagreGraph.setNode(node.id, {
      width: node.width,
      height: node.height
    });
  });
```

After layouting with dagre, you can use the width and height to get the positions for React Flow:

```
    node.position = {
      x: nodeWithPosition.x - node.width / 2,
      y: nodeWithPosition.y - node.height / 2
    };
```

Another option would be to create an invisible wrapper around your node which has the same size as the other nodes, then centering the smaller node inside that wrapper.

**Comment by moklick** [comment]
Please re-open if the issue still persists.
